File: 200/223 - Rectangle Area.java
public class Solution {
    public int computeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
        long sum = ((long)C - (long)A) * ((long)D - (long)B) + ((long)G - (long)E) * ((long)H - (long)F);
        long overlap = Math.max(Math.min((long)C, (long)G) - Math.max((long)A, (long)E), 0) * Math.max(Math.min((long)D, (long)H) - Math.max((long)B, (long)F), 0);
        
        return Math.toIntExact(sum - overlap);
    }
}

File: csharp/223.rectangle-area.cs
/*
 * @lc app=leetcode id=223 lang=csharp
 *
 * [223] Rectangle Area
 */
public class Solution {
    public int ComputeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
        int a1 = (C - A) * (D - B), a2 = (G - E) * (H - F);
        if (A >= G || B >= H || E >= C || F >= D) {
            return a1 + a2;
        }
        return a1 + a2 - (Math.Min(C, G) - Math.Max(A, E)) * (Math.Min(D, H) - Math.Max(B, F));
    }
}

